src/log.rs

```diff
@@ -61,38 +60,94 @@ macro_rules! emit_event {
     };
 }
 
-pub trait LogExt<T> {
+pub trait LogExt<T, E>
+where
+    Self: std::marker::Sized,
+{
+    #[track_caller]
+    fn format_log(self, context: &Context, msg: Option<&str>) -> Result<T, E>;
+
     /// Emits a warning if the receiver contains an Err value.
     ///
-    /// Returns an [`Option<T>`] with the `Ok(_)` value, if any:
-    /// - You won't get any warnings about unused results but can still use the value if you need it
-    /// - This prevents the same warning from being printed to the log multiple times
-    ///
     /// Thanks to the [track_caller](https://blog.rust-lang.org/2020/08/27/Rust-1.46.0.html#track_caller)
     /// feature, the location of the caller is printed to the log, just like with the warn!() macro.
+    ///
+    /// Unfortunately, the track_caller feature does not work on async functions (as of Rust 1.50).
+    /// Once it is, you can add `#[track_caller]` to helper functions that use one of the log helpers here
+    /// so that the location of the caller can be seen in the log. (this won't work with the macros,
+    /// like warn!(), since the file!() and line!() macros don't work with track_caller)  
+    /// See https://github.com/rust-lang/rust/issues/78840 for progress on this.
```

Comment:
I don't really understand what all this is trying to tell me btw.  I mean, these functions here that have track_caller aren't async so why the long explanation around this?  We can call them from inside async functions can we not?

Reply:
>  We can call them from inside async functions can we not?

Yes, but I had hoped to be able to mark some often-used helper functions like `get_config()` with #[track_caller] so that we get the caller's location, which usually is more interesting.

This comment was rather meant to prevent others from trying this out and failing themselves :)